src/services/dynamicsDal.service.js

```diff
@@ -18,99 +18,103 @@ module.exports = class DynamicsDalService {
-    return new Promise((resolve, reject) => {
-      // Combine the path to Dynamics and the query and add it to our request options
-      this.crmRequestOptions.path = `${config.dynamicsWebApiPath}${query}`
+  _setCallSpecificRequestOptions (query, method, dataObject = undefined) {
+    // Combine the path to Dynamics and the query and add it to our request options
+    this.crmRequestOptions.path = `${config.dynamicsWebApiPath}${query}`
+    this.crmRequestOptions.method = method
 
-      this.crmRequestOptions.method = 'GET'
+    // Set the content length
+    if (dataObject) {
+      const contentLength = Buffer.byteLength(JSON.stringify(dataObject))
+      this.crmRequestOptions.headers['Content-Length'] = contentLength
+    }
+  }
 
+  _call (query, dataObject = undefined) {
+    return new Promise((resolve, reject) => {
       // Query Dynamics for the data via a HTTPS request
       const crmRequest = https.request(this.crmRequestOptions, function (response) {
+        response.setEncoding('utf8')
+
         // We use an array to hold the response parts in the event we get multiple parts returned
         const responseParts = []
-        response.setEncoding('utf8')
         response.on('data', function (chunk) {
           responseParts.push(chunk)
         })
         response.on('end', function () {
           console.log(`Dynamics query response: Status Code: ${response.statusCode} Message: ${response.statusMessage}`)
 
-          if (response.statusCode === 200) {
-            // Parse the response JSON
-            resolve(JSON.parse(responseParts.join('')))
-          } else {
-            console.error(responseParts.join(''))
-            const message = `Unknown response from Dynamics. Code: ${response.statusCode} Message: ${response.statusMessage}`
-            reject(message)
+          switch (response.statusCode) {
+            case 200:
+              // Parse the response JSON
+              resolve(JSON.parse(responseParts.join('')))
+              break
+            case 204:
+              resolve(response.headers['odata-entityid'])
+              break
+            default:
+              const message = `Unknown response from Dynamics. Code: ${response.statusCode} Message: ${response.statusMessage}`
+              reject(message)
           }
         })
       })
-
       crmRequest.on('error', function (error) {
         console.error('Dynamics error: ' + error)
         reject(error)
       })
 
+      // Write the data
+      if (dataObject) {
+        crmRequest.write(JSON.stringify(dataObject))
+      }
+
       // Close the Web Api request
       crmRequest.end()
     })
   }
 
-  _commit (query, method, dataObject = undefined) {
-    return new Promise((resolve, reject) => {
-      // Combine the path to Dynamics and the query and add it to our request options
-      this.crmRequestOptions.path = `${config.dynamicsWebApiPath}${query}`
+  /*
+    Extracts id of an entity from the value Dyanmics returns
```

Comment:
This extremely detailed explanation is awesome (but there's a typo in this line 🙂)